src/urllib3/poolmanager.py

```diff
@@ -449,14 +496,14 @@ class ProxyManager(PoolManager):
 
     def __init__(
         self,
-        proxy_url,
-        num_pools=10,
-        headers=None,
-        proxy_headers=None,
+        proxy_url: str,
+        num_pools: int = 10,
+        headers: Optional[Mapping[str, str]] = None,
+        proxy_headers: Optional[Mapping[str, str]] = None,
         proxy_ssl_context=None,
```

Comment:
`Optional["ssl.SSLContext"]`?

Reply:
Then we get ` Name 'ssl' is not defined` from mypy

Reply:
Can have `ssl` import within a `if typing.TYPE_CHECKING:` branch